client/core/core.go

```diff
@@ -4973,11 +4997,14 @@ func handlePreimageRequest(c *Core, dc *dexConnection, msg *msgjson.Message) err
 	return nil
 }
 
-func processPreimageRequest(c *Core, dc *dexConnection, reqID uint64, oid order.OrderID) error {
+func processPreimageRequest(c *Core, dc *dexConnection, reqID uint64, oid order.OrderID, commitChecksum dex.Bytes) error {
 	tracker, preImg, isCancel := dc.findOrder(oid)
 	if tracker == nil {
 		return fmt.Errorf("no active order found for preimage request for %s", oid)
 	}
+
+	tracker.csum = commitChecksum
+
```

Comment:
We've forgotten about cancel orders, which are not `trackedTrades` themselves, but the `trackedTrade.cancel` field that links it with the targeted trade.  Please see the following suggested diff: https://github.com/decred/dcrdex/commit/cbf49a8512913b4683e611e2d1c272a068b9e64f

I'd also prefer the `trackedTrade` formatted like in that diff too if you don't mind applying that too.

Reply:
Okay, I've applied your diff, thanks! Looks like it addresses all the concerns I've outlined above - https://github.com/decred/dcrdex/pull/1077#discussion_r628744407

Reply:
I should just note as a follow-up to the other discussion in this PR w.r.t. mutex/atomic that I felt using the `trackedTrade.mtx` was alright in the end because we had to safely access the `cancel` field as well.